src/SFA.DAS.Commitments.Application/Commands/CohortApproval/ProiderApproveCohort/ProviderApproveCohortCommandHandler.cs

```diff
@@ -56,82 +50,18 @@ protected override async Task HandleCore(ProviderApproveCohortCommand message)
-            }
-        }
-
-        private async Task PublishApprovalRequestedMessage(Commitment commitment)
-        {
-            await _messagePublisher.PublishAsync(new CohortApprovalRequestedByProvider(commitment.EmployerAccountId, commitment.ProviderId.Value, commitment.Id));
-        }
-
-        private async Task UpdateCommitment(Commitment commitment, bool haveBothPartiesApproved, string userId, string lastUpdatedByName, string lastUpdatedByEmail, string message)
-        {
-            var updatedEditStatus = DetermineNewEditStatus(haveBothPartiesApproved);
-            var changeType = _cohortApprovalService.DetermineHistoryChangeType(haveBothPartiesApproved);
-            _historyService.TrackUpdate(commitment, changeType.ToString(), commitment.Id, null, CallerType.Provider, userId, commitment.ProviderId, commitment.EmployerAccountId, lastUpdatedByName);
-
-            commitment.EditStatus = updatedEditStatus;
-            commitment.LastAction = LastAction.Approve;
-            commitment.CommitmentStatus = CommitmentStatus.Active;
-            commitment.LastUpdatedByProviderEmail = lastUpdatedByEmail;
-            commitment.LastUpdatedByProviderName = lastUpdatedByName;
             
-            if (haveBothPartiesApproved && !commitment.HasTransferSenderAssigned)
+            var userInfo = new UserInfo
             {
-                var account = await _employerAccountsService.GetAccount(commitment.EmployerAccountId);
-
-                commitment.ApprenticeshipEmployerTypeOnApproval = account.ApprenticeshipEmployerType;
-            }
-            
-            await Task.WhenAll(
-                _cohortApprovalService.AddMessageToCommitment(commitment, lastUpdatedByName, message, CallerType.Provider),
-                _commitmentRepository.UpdateCommitment(commitment),
-                _historyService.Save()
-            );
-        }
+                UserDisplayName = message.LastUpdatedByName,
+                UserEmail = message.LastUpdatedByEmail,
+                UserId = message.UserId
+            };
 
-        private EditStatus DetermineNewEditStatus(bool haveBothPartiesApproved)
-        {
-            return haveBothPartiesApproved ? EditStatus.Both : EditStatus.EmployerOnly;
-        }
-
-        private static AgreementStatus DetermineNewAgreementStatus(bool haveBothPartiesApproved)
-        {
-            var newAgreementStatus = haveBothPartiesApproved ? AgreementStatus.BothAgreed : AgreementStatus.ProviderAgreed;
-            return newAgreementStatus;
-        }
-
-        private bool HaveBothPartiesApproved(Commitment commitment)
-        {
-            var currentAgreementStatus = _cohortApprovalService.GetCurrentAgreementStatus(commitment);
-            return currentAgreementStatus == AgreementStatus.EmployerAgreed;
+            await _v2EventsPublisher.SendProviderApproveCohortCommand(message.CommitmentId, message.Message, userInfo);
```

Comment:
could it possible for this line to throw a null reference exception as the IV2EventsPublisher appears to be an optional parameter?